README.md

```diff
@@ -1577,38 +1604,63 @@ worker = Worker(
 For plugins that need to work with both clients and workers, you can implement both interfaces in a single class:
 
 ```python
+import temporalio
+from contextlib import AbstractAsyncContextManager
+from typing import AsyncIterator
 from temporalio.client import Plugin as ClientPlugin, ClientConfig
-from temporalio.worker import Plugin as WorkerPlugin, WorkerConfig
+from temporalio.worker import Plugin as WorkerPlugin, WorkerConfig, ReplayerConfig, Worker, Replayer, WorkflowReplayResult
 
 
 class UnifiedPlugin(ClientPlugin, WorkerPlugin):
-  def configure_client(self, config: ClientConfig) -> ClientConfig:
-    # Client-side customization
-    config["namespace"] = "unified-namespace"
-    return super().configure_client(config)
-
-  def configure_worker(self, config: WorkerConfig) -> WorkerConfig:
-    # Worker-side customization
-    config["max_cached_workflows"] = 500
-    return super().configure_worker(config)
-
-  async def run_worker(self, worker: Worker) -> None:
-    print("Starting unified worker")
-    await super().run_worker(worker)
-
+    def init_client_plugin(self, next: ClientPlugin) -> None:
+        self.next_client_plugin = next
 
+    def init_worker_plugin(self, next: WorkerPlugin) -> None:
+        self.next_worker_plugin = next
+  
+    def configure_client(self, config: ClientConfig) -> ClientConfig:
+        # Client-side customization
+        config["data_converter"] = pydantic_data_converter
+        return self.next_client_plugin.configure_client(config)
+  
+    async def connect_service_client(
+        self, config: temporalio.service.ConnectConfig
+    ) -> temporalio.service.ServiceClient:
+        # Add authentication to the connection
+        config.api_key = self.api_key
+        return await self.next_client_plugin.connect_service_client(config)
+        
+    def configure_worker(self, config: WorkerConfig) -> WorkerConfig:
+        # Worker-side customization
+        return self.next_worker_plugin.configure_worker(config)
+  
+    async def run_worker(self, worker: Worker) -> None:
+        print("Starting unified worker")
+        await self.next_worker_plugin.run_worker(worker)
+        
+    def configure_replayer(self, config: ReplayerConfig) -> ReplayerConfig:
+        config["data_converter"] = pydantic_data_converter
+        return config
```

Comment:
@tconley1428 This should call `self.next_worker_plugin.configure_replayer` right?